### PR TITLE
Fix problem with SIGSEGV on FreeBSD 12+

### PR DESCRIPTION
**Pull request by girgen**
This is partly related to https://github.com/fluent/fluent-bit/pull/2463 for MacOS.

The first one is in  flb_pipe_r() function returning 0 on FreeBSD 12+ and that
brings up errors similar to "[error] [src/flb_scheduler.c:52 errno=0] No error:
0". This patch will fix it the same way as it was fixed by upstream code for
MacOS.

The second issue is a SIGSEGV that is caused by incorrect section
structure of the resulting binary file generated by clang due to some gcc
specific code in libco.  Since there are two copies of libco (flb_libco) code
in the fluent-bit sources - one in the lib/flb_libco folder and another one in
lib/monkey/deps/flb_libco - they are both patched the same way.

This patches were tested on FreeBSD 11.4 and 12.2 amd64.
The resulting binaries was tested with "elfdump -a" to contain only one ".text"
section and their startup logs were tested running "fluent-bit -v -i dummy -o
stdout".

Submitted by: Yuri Pankov and Artyom Davidov
Signed-off-by: Palle Girgensohn <girgen@FreeBSD.org>

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

**Documentation**

- [ ] Documentation required for this feature

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by girgen** [comment]
Isssue described and discussed here: https://github.com/fluent/fluent-bit/issues/2747

File: lib/monkey/deps/flb_libco/settings.h [hunk 1]
```diff
@@ -42,6 +42,8 @@
   #define text_section __declspec(allocate(".text"))
 #elif defined(__APPLE__) && defined(__MACH__)
   #define text_section __attribute__((section("__TEXT,__text")))
+#elif defined(__clang__)
+  #define text_section __attribute__((section(".text")))
```
Comment (fujimotos) [inline lib/monkey/deps/flb_libco/settings.h:46]: Can you extract this changeset into a separate patch, and submit
it to https://github.com/edsiper/flb_libco?

Our standard workflow for a library fix is to pull the patch in the upstream repo
and bring it to the downstream. So we prefer getting this patch there.

**Comment by fujimotos** [comment]
The DCO checker is failing on f88053c9. This is probably because
the commit is certificated by

> Signed-off-by: Palle Girgensohn <girgen@FreeBSD.org>

... while commit is authored by:

> From: Palle Girgensohn <girgen@pingpong.net>

@girgen Can you fix the commit metadata and re-push it so that the commit
author matches with the developer certificate of origin?

**Comment by girgen** [comment]
ok. I push a new version here and a separate fix for lib_co.

**Review by fujimotos — APPROVED** [review]
@girgen Can you fix the title line of the commit title from:

> Fix problem with SIGSEGV on FreeBSD 12+ 

to:

> scheduler: Fix problem with SIGSEGV on FreeBSD 12+ 

After that, I believe this patch is commitable.

**Comment by edsiper** [comment]
Since I am making a new release now I will fix the commit subject to include the fix.

**Comment by edsiper** [comment]
thanks!
